atlas_anndata/anndata_ops.py

```diff
@@ -171,7 +171,7 @@ def overwrite_obs_with_magetab(
     return adata
 
 
-def derive_metadata(adata, config, kind=None):
+def derive_metadata(adata, config, write_premagetab, kind=None):
```

Comment:
My suggestion is to use `True` or `False` for the value of `write_premagetab` since it is used as a boolean.  If writing premagetab is the expected default action, set it in the function as `write_premagetab=True` (also so that scripts with the older version of the function don't fail), and then write `write_premagetab=False` explicitly when needed.